go-controller/pkg/ovnwebhook/podadmission.go

```diff
@@ -0,0 +1,167 @@
+package ovnwebhook
+
+import (
+	"context"
+	"fmt"
+
+	"golang.org/x/exp/maps"
+
+	corev1 "k8s.io/api/core/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	listers "k8s.io/client-go/listers/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kubevirt"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+// checkPodAnnot defines an additional checks for the allowed annotations
+type checkPodAnnot func(nodeLister listers.NodeLister, v annotationChange, pod *corev1.Pod, nodeName string) error
+
+// interconnectPodAnnotationChecks holds annotations allowed for ovnkube-node:<nodeName> users in IC environments
+var interconnectPodAnnotations = map[string]checkPodAnnot{
+	util.OvnPodAnnotationName: func(nodeLister listers.NodeLister, v annotationChange, pod *corev1.Pod, nodeName string) error {
```

Comment:
should we error for remote pods here? after the kubevirt check  I mean.. if the pod is not local and you try to do something to it, its a clear error, so we could just ignore not scheduled on this node pods right?

Reply:
This is already checked earlier, ovnkube-node is not allowed to modify pods that are not local to the node.